moto/sagemaker/models.py

```diff
@@ -1243,6 +1246,108 @@ def get_training_job_tags(self, arn):
         except RESTError:
             return []
 
+    def list_training_jobs(
+        self,
+        next_token,
+        max_results,
+        creation_time_after,
+        creation_time_before,
+        last_modified_time_after,
+        last_modified_time_before,
+        name_contains,
+        status_equals,
+        sort_by=None,
+        sort_order=None,
+    ):
+        if sort_by is None:
+            sort_by = "CreationTime"
+        if sort_order is None:
+            sort_order = "Ascending"
+
+        if next_token:
+            try:
+                starting_index = int(next_token)
+                if starting_index > len(self.training_jobs):
+                    raise ValueError  # invalid next_token
+            except ValueError:
+                raise AWSValidationException('Invalid pagination token because "{0}".')
+        else:
+            starting_index = 0
+
+        if max_results:
```

Comment:
The limiting should probably be done at the very end, after filtering the results for name, creation time, etc.

Reply:
I believe limiting is the first step, unless I'm reading the [aws documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.list_training_jobs) wrong.

`When StatusEquals and MaxResults are set at the same time, the MaxResults number of training jobs are first retrieved ignoring the StatusEquals parameter and then they are filtered by the StatusEquals parameter, which is returned as a response.`

Reply:
Ah, you're right. Just did some testing, and it looks like the other params behave the same as well (CreationTimeBefore etc). That's unusual - good catch!